From 8067d2bce8820f2c680e98b637c6315b75566260 Mon Sep 17 00:00:00 2001
From: Francois Perrad <francois.perrad@gadz.org>
Date: Mon, 8 Nov 2010 18:30:19 +0100
Subject: [PATCH] implement setfenv (deprecated with Lua 5.2.0 work3) with debug library

---
 src/Coat.lua |   18 +++++++++++++++++-
 1 files changed, 17 insertions(+), 1 deletions(-)

diff --git a/src/Coat.lua b/src/Coat.lua
index b98b8ae..bb103d0 100644
--- a/src/Coat.lua
+++ b/src/Coat.lua
@@ -10,7 +10,6 @@ local pcall = pcall
 local rawget = rawget
 local rawset = rawset
 local require = require
-local setfenv = setfenv
 local setmetatable = setmetatable
 local tostring = tostring
 local basic_type = type
@@ -25,6 +24,23 @@ local _M = {}
 
 local Meta = require 'Coat.Meta.Class'
 
+local function setfenv (level, M)
+    local func = debug.getinfo(level + 1, 'f').func
+    local up = 1
+    while true do
+        local name = debug.getupvalue(func, up)
+        if name == '_ENV' then -- Lua 5.2
+            debug.setupvalue(func, up, M)
+            return
+        end
+        if name == nil then
+            break
+        end
+        up = up + 1
+    end
+    debug.setfenv(func, M) -- Lua 5.1
+end
+
 local function type (obj)
     local t = basic_type(obj)
     if t == 'table' then
-- 
1.7.1

